### PR TITLE
[Prim]Support mean_grad decompose in vjp

### PR DESCRIPTION
**Pull request by Aurelius84**
### PR Category

Operator Mechanism

### PR Types

New features

### Description

Pcard-67164

### CODE REVIEW

File: paddle/fluid/primitive/rule/vjp/details.h [hunk 1]
```diff
@@ -154,6 +154,60 @@ void sum_grad(const Tensor& x,
...
+      for (size_t i = 0; i < x_dim.size(); ++i) {
+        axis_data.push_back(i);
+      }
+    }
+    bool switch_dynamic = false;
+    int64_t factor = 1;
+    for (int64_t idx : axis_data) {
+      if (idx < 0) idx += x_dim.size();
+      if (x_dim[idx] == -1) {
+        switch_dynamic = true;
```
Comment (cyber-pioneer) [inline paddle/fluid/primitive/rule/vjp/details.h]: 建议用 Paddle/paddle/fluid/primitive/utils/utils.h 里的has_dynamic_shape 来判断动态shape

File: paddle/fluid/primitive/codegen/gen.py [hunk 2]
```diff
@@ -96,6 +96,7 @@
     'gather_nd_grad',
     'pad_grad',
     'prod_grad',
+    'mean_grad',
```
Comment (cyber-pioneer) [inline paddle/fluid/primitive/codegen/gen.py]: 放在 CUSTOM_VJP里

File: paddle/fluid/primitive/rule/vjp/details.h [hunk 3]
```diff
@@ -154,6 +154,60 @@ void sum_grad(const Tensor& x,
...
+               const Tensor& out_grad,
+               const IntArray& axis,
+               bool keepdim,
+               bool reduce_all,
+               Tensor* x_grad) {
+  if (!x_grad) {
+    return;
+  }
+  Tensor x_grad_tmp;
+  sum_grad<T>(x, out_grad, axis, keepdim, reduce_all, &x_grad_tmp);
```
Comment (cyber-pioneer) [inline paddle/fluid/primitive/rule/vjp/details.h:168]: sum_grad 尚未适配动态shape，

Comment (Aurelius84) [reply paddle/fluid/primitive/rule/vjp/details.h:168]: 后续需统一考虑支持

File: paddle/fluid/primitive/rule/vjp/details.h (hunk 1 above)
Comment (Aurelius84) [reply paddle/fluid/primitive/rule/vjp/details.h]: 已放到utils复用

**Review by cyber-pioneer — APPROVED** [review]

**Pull request merged**